akka-stream/src/main/scala/akka/stream/impl/StreamLayout.scala

```diff
@@ -283,14 +291,21 @@ private[akka] object StreamLayout {
 
     override def toString =
       s"""
-        | Module: ${this.attributes.nameOption.getOrElse("unnamed")}
-        | Modules: ${subModules.toSeq.map(m ⇒ "   " + m.attributes.nameOption.getOrElse(m.getClass.getName)).mkString("\n")}
+        | Module: ${this.attributes.nameOrDefault("unnamed")}
+        | Modules: ${subModules.iterator.map(m ⇒ "   " + m.attributes.nameOrDefault(m.getClass.getName)).mkString("\n")}
         | Downstreams:
-        | ${downstreams.map { case (in, out) ⇒ s"   $in -> $out" }.mkString("\n")}
+        | ${downstreams.iterator.map { case (in, out) ⇒ s"   $in -> $out" }.mkString("\n")}
         | Upstreams:
-        | ${upstreams.map { case (out, in) ⇒ s"   $out -> $in" }.mkString("\n")}
+        | ${upstreams.iterator.map { case (out, in) ⇒ s"   $out -> $in" }.mkString("\n")}
       """.stripMargin
 
+    override def describe: String =
+      attributes.nameOrDefault(null) match {
+        case null ⇒ "(" + subModules.iterator.map(_.describe).mkString(" ~> ") + ")"
```

Comment:
This is definitely wrong, printing the structure will require proper interpretation of all the wiring inside of it.